src/embed.ts

```diff
@@ -244,18 +244,32 @@ async function _embed(
   const logLevel = opts.logLevel ?? vega.Warn;
   const downloadFileName = opts.downloadFileName ?? 'visualization';
 
+  const div = typeof el === 'string' ? document.querySelector(el) : el;
+  if (!div) {
+    throw Error(`${el} does not exist`);
+  }
+
   if (opts.defaultStyle !== false) {
     // Add a default stylesheet to the head of the document.
     const ID = 'vega-embed-style';
-    if (!document.getElementById(ID)) {
+    const possibleRoot = div.getRootNode ? div.getRootNode() : document;
+    let root: ShadowRoot | Document;
```

Comment:
Can you extract this into a function `const {root, rootContainer} = getRoots(possibleRoot)`?